charts/spire/charts/spire-server/templates/statefulset.yaml

```diff
@@ -238,6 +241,16 @@ spec:
         {{- if gt (len .Values.extraVolumes) 0 }}
         {{- toYaml .Values.extraVolumes | nindent 8 }}
         {{- end }}
+        {{- if eq .Values.persistence.type "emptyDir" }}
+        - name: spire-data
+          emptyDir: {}
+        {{- else if eq .Values.persistence.type "hostPath" }}
+        - name: spire-data
+          hostPath:
+            path: {{ .Values.persistence.hostPath }}
+            type: Directory
+        {{- end }}
+  {{- if eq .Values.persistence.type "pvc" }}
```

Comment:
Please make this key off the "if-elseif-end" above, making it an "if-elseif-elseif-end" OR make each statement an "if-end" statement.  By having two values handled one way and the third a different way, it presents more mental load on developers reading it.

I'm fine including each block independently keyed of its own activating value.  I'm also ok with a giant switch statement implemented in if-endif-endif-else patterns.  I'm just not fine with the hybrid of the two on one enum.  If you care to know my preference, it would be for multiple if-end statements, as they can be moved more easily.

Reply:
Since the volume mounts and volumes are different blocks, different if's for those blocks makes more sense to me. Then I think the separate if's for the items in volumeMounts would be the better route. What do others think?

Reply:
Do we really need a group consensus for a refactoring call?  If it altered functionality, sure; but, this seems to be more a call to ask if we can create contention on a relatively minor review point.

Reply:
This pr is already blocked. Just a bit hesitent to continue to work on it if its going to get killed off. If it opens back up, then yeah, I can make the requested changes.

Reply:
@kfox1111 Please don't read into a request that the work is going to be killed off.  Prompt handling of requests is the main reason submissions get merged.

Yes, we need to work on the team's review flexibility, occasionally we all get stuck in items that really don't matter, as we are all emotional creatures inclined to align to our first impressions (right or wrong).  That said, if you lose heart, you will make your failure to get the change merged a self-fulfilling prophecy.

Please give the effort an honest amount of work.  If you really feel you are losing ground on an item, then say so explicitly, asking for the changes be limited to those that really matter, or for the review to be closed as not accepted.  But, I would hope that you don't reach for that option so soon.  In this case, I see this as merging, and most of my blocking questions are resolved now.